google/cloud/internal/rest_carrier.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+#include "google/cloud/internal/rest_carrier.h"
+#include "google/cloud/internal/rest_context.h"
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+opentelemetry::nostd::string_view RestCarrier::Get(
+    opentelemetry::nostd::string_view) const noexcept {
+  // Since the client is never extracting data from the REST headers, we are not
+  // implementing Get and returning an empty string by default.
+  return "";
+}
+
+void RestCarrier::Set(opentelemetry::nostd::string_view key,
+                      opentelemetry::nostd::string_view value) noexcept {
```

Comment:
Argh, I guess we should be using `internal::NoExceptAction`. (Also in the gRPC carrier too :disappointed: )

https://github.com/googleapis/google-cloud-cpp/blob/91526ded74e602f3d47d603b69d9f231f40657b3/google/cloud/internal/noexcept_action.h#L33

Reply:
https://github.com/googleapis/google-cloud-cpp/issues/12400